### DISCUSSION TITLE
[plugin-sitemap] Is it possible to generate a custom sitemap using frontMatter?

### DISCUSSION THREAD

**Discussion by 3Alan** [facebook/docusaurus#8316] [Q&A]
For example, generate the lastest 10 blogs sitemap by date in frontMatter.

**Comment by slorber** [comment]
We don't provide that option.

In 3.x (soon) you can mark a blog as `unlisted` and it's excluded from sitemaps (but also UI/search/google...).

Having a frontmatter to exclude from sitemap one by one could make sense.

In any case you can create your own sitemaps plugin if the default one is not good enough

**Comment by 3Alan** [comment]
I'm not  sure if we can add a option in the default `plugin-sitemap`, like `getRoutesPath(postBuildProps)`, if user pass this option, plugin use the getRoutesPath return value as source data, then user can customize sitemap by pass `options.getRoutesPath`
```js
async postBuild(props) {
  const {siteConfig, routesPaths, outDir, head} = props;
  if (siteConfig.noIndex) {
    return;
  }
  const routes = options.getRoutesPath ? options.getRoutesPath(props) : routesPaths;

  // Generate sitemap.
  const generatedSitemap = await createSitemap(
    siteConfig,
    routes,
    head,
    options,
  );

 ...
},
```

If not, maybe I should create a new sitemap plugin

**Reply by slorber** [reply]
That could work yes but at the same time we already have an ignorePatterns options: https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-sitemap#ignorePatterns
Patterns doc: https://github.com/micromatch/micromatch

Isn't this good enough for your use-case?

Maybe `"!(/blog/post1|/blog/post2)"`

We can as well add an `includePatterns` option to make it simpler

**Reply by 3Alan** [reply]
Sorry, my english is not good.
Yeah, I know the `ignorePatterns` options, but it not enough for my use-case. 
My use-case is:
I use Google Indexing API  notify Google when my pages are added or updated, so I need 2 sitemaps, one containing all pages for Google crawlers and one containing only the most recently updated pages for Google Indexing API.

**Reply by slorber** [reply]
You can use the sitemaps plugin twice:
- one for all the pages
- one with a more aggressive `ignorePatterns` option to only include the ones you want

Now the question is: how do you plug your own algo to include only the most recent ones. Is the `options.getRoutesPath(props)` data we provide enough in your case? How would you implement the logic exactly?

Please explain why you want 2 sitemaps exactly. Why is it important? Is this a current SEO practice? Do you have good reasons to do so that can be backed by any reference link from an authority like Google documentation?

**Reply by 3Alan** [reply]
Oh, my bad, thanks for the correction, I shouldn't have used the sitemap for other purposes. After thinking about it, I really don't need 2 sitemaps, all I need is the url of the most recently updated blog to submit to Google, so I would create a plugin to generate a json file containing the most recently updated blog, like this

```js
const fs = require('fs-extra');
const path = require('path');
const logger = require('@docusaurus/logger');

module.exports = function pluginIndexingUrls() {
  return {
    name: 'docusaurus-plugin-indexing-urls',

    async postBuild({ siteConfig, plugins, outDir }) {
      if (siteConfig.noIndex) {
        return;
      }

      const contentBlogPluginContext = plugins.find(
        item => item.name === 'docusaurus-plugin-content-blog'
      );
      const { blogPosts } = contentBlogPluginContext.content;
      const sortedBlogPosts = blogPosts.sort(
        (a, b) => Date.parse(a.metadata.date) - Date.parse(b.metadata.date)
      );

      const recentBlogPostsPath = sortedBlogPosts
        .map(item => item.metadata.permalink)
        .slice(0, 10);

      if (recentBlogPostsPath.length === 0) {
        return;
      }

      // Write json file.
      const indexingUrlsPath = path.join(outDir, 'indexing-urls.json');
      try {
        await fs.outputFile(
          indexingUrlsPath,
          JSON.stringify(recentBlogPostsPath)
        );
      } catch (err) {
        logger.error('Writing indexing urls file failed.');
        throw err;
      }
    }
  };
};
```

and why I need the latest updated url is a reference to [Google documentation](https://developers.google.com/search/apis/indexing-api/v3/quickstart#sitemaps)
